compiler/rustc_codegen_llvm/src/va_arg.rs

```diff
@@ -173,26 +173,26 @@ pub(super) fn emit_va_arg(
     // is lacking in some instances, so we should only use it as a fallback.
     let target = &bx.cx.tcx.sess.target;
     let arch = &bx.cx.tcx.sess.target.arch;
-    match (&**arch, target.options.is_like_windows) {
+    match (&**arch, target.options.is_like_windows, target.options.is_like_osx) {
         // Windows x86
-        ("x86", true) => {
+        ("x86", true, _) => {
             emit_ptr_va_arg(bx, addr, target_ty, false, Align::from_bytes(4).unwrap(), false)
         }
         // Generic x86
-        ("x86", _) => {
+        ("x86", _, _) => {
             emit_ptr_va_arg(bx, addr, target_ty, false, Align::from_bytes(4).unwrap(), true)
         }
         // Windows AArch64
-        ("aarch64", true) => {
+        ("aarch64", true, _) => {
             emit_ptr_va_arg(bx, addr, target_ty, false, Align::from_bytes(8).unwrap(), false)
         }
-        // iOS AArch64
-        ("aarch64", _) if target.target_os == "ios" => {
+        // macOS/iOS AArch64
+        ("aarch64", _, true) => {
```

Comment:
Minor style nit: matching on forever increasing tuple elements seems untenable in the future. It seems to me like `if target.options.is_like_osx` would result in slightly more maintainable code.

Reply:
Would you like me to change the windows one as well?

Reply:
Entirely your call, but I believe doing so for all kinds of effective guards will produce nicer code overall.

Reply:
Did this in a first commit.